index-scheduler/src/lib.rs

```diff
@@ -256,6 +260,9 @@ pub struct IndexScheduler {
     /// Store the tasks associated to an index.
     pub(crate) index_tasks: Database<Str, RoaringBitmapCodec>,
 
+    /// Store the tasks that were canceled by a task uid
+    pub(crate) canceled_by: Database<OwnedType<BEU32>, RoaringBitmapCodec>,
```

Comment:
Nice, you added a database where the key is the task-cancelation id, and the value is a roaring bitmap with the tasks it canceled. That's cool ❤️

Reply:
haha, yes you were completely right about needing another database :)